Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome

The user @topas-rec reported in https://github.com/bit-team/backintime/issues/1421#issuecomment-1509761121 a failing unit test on ARCH Linux caused by this additional (not expected) warning in the actual output:

> Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.

Furthermore this warning occurs if the optional package `qt5-wayland` is not installed:

> qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""

### System info:

```
$ echo $XDG_SESSION_TYPE
wayland

$ echo $QT_QPA_PLATFORM


$ echo $XDG_CURRENT_DESKTOP
GNOME
```

### Details

```
python -m unittest  -b test/test_backintime.py
.Fs
======================================================================
FAIL: test_local_snapshot_is_successful (test.test_backintime.TestBackInTime)
end to end test - from BIT initialization through snapshot
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tobiasb/.cache/yay/backintime/src/backintime-1.3.3/common/test/test_backintime.py", line 147, in test_local_snapshot_is_successful
    self.assertRegex(filtered_log_output, re.compile(r'''INFO: Lock
AssertionError: Regex didn't match: 'INFO: Lock\nINFO: Take a new snapshot. Profile: 1 Main profile\nINFO: Call rsync to take the snapshot\nINFO: Save config file\nINFO: Save permissions\nINFO: Create info file\nINFO: Unlock\n' not found in 'INFO: Lock\nINFO: Take a new snapshot. Profile: 1 Main profile\nINFO: Call rsync to take the snapshot\nINFO: Save config file\nINFO: Save permissions\nINFO: Create info file\nWarning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.\nqt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""\nThis application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.\n\nAvailable platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.\n\nINFO: Unlock\n'
```

### See also

https://aur.archlinux.org/pkgbase/backintime#comment-911755

using git PKGBUILD [backintime-git](https://aur.archlinux.org/packages/backintime-git) still fails on testing:
make: *** [Makefile:570: unittest] Error 1
```
Traceback (most recent call last):
  File "/home/joe/.cache/yay/backintime-git/src/backintime/common/test/test_backintime.py", line 176, in test_local_snapshot_is_successful
    self.assertRegex(filtered_log_output, re.compile(r'''INFO: Lock
AssertionError: Regex didn't match: 'INFO: Lock\nINFO: Take a new snapshot. Profile: 1 Main profile\nINFO: Call rsync to take the snapshot\nINFO: Save config file\nINFO: Save permissions\nINFO: Create info file\nINFO: Unlock' not found in 'INFO: Lock\nINFO: Take a new snapshot. Profile: 1 Main profile\nINFO: Call rsync to take the snapshot\nINFO: Save config file\nINFO: Save permissions\nWarning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.\nINFO: Create info file\nINFO: Unlock'
```

--- a/CHANGES
+++ b/CHANGES
@@ -3,6 +3,7 @@ Back In Time
 Version 1.3.4-dev (development of upcoming release)
 * Project: Renamed branch "master" to "main" and started "gitflow" branching model.
 * Fix bug: Add support for ChainerBackend class as keyring which iterates over all supported keyring backends (#1410)
+# Fix bug: Unit test fails on some machines due to warning "Ignoring XDG_SESSION_TYPE=wayland on Gnome..." (#1429)
 
 Version 1.3.3 (2023-01-04)
 * New feature: Command line argument "--diagnostics" to show helpful info for better issue support (#1100)

--- a/common/test/test_backintime.py
+++ b/common/test/test_backintime.py
@@ -145,10 +145,12 @@ under certain conditions; type `backintime --license' for details.
             "WARNING: D-Bus message:",
             "WARNING: Udev-based profiles cannot be changed or checked",
             "WARNING: Inhibit Suspend failed",
+            "Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway"
         ]
 
         line_contains_to_exclude = [
-            "Gtk-WARNING"
+            "Gtk-WARNING",
+            "qt.qpa.plugin: Could not find the Qt platform plugin"
         ]
 
         # remove lines via startswith()

